src/libraries/System.Text.Json/System.Text.Json.SourceGeneration.UnitTests/JsonSourceGeneratorTests.cs

```diff
@@ -143,40 +138,85 @@ public class NotMyType { }
-                MetadataReference.CreateFromFile(typeof(KeyValuePair).Assembly.Location),
-                MetadataReference.CreateFromFile(systemRuntimeAssemblyPath),
-            };
-
-            return CSharpCompilation.Create(
-                "TestAssembly",
-                syntaxTrees: new[] { CSharpSyntaxTree.ParseText(source) },
-                references: references,
-                options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
-            );
-        }
+            string source = @"
+            using System;
+            using System.Text.Json.Serialization;
+            using JsonSerializable = System.Text.Json.Serialization.JsonConstructorAttribute;
+            using AliasedAttribute = System.Text.Json.Serialization.JsonSerializableAttribute;
+
+              namespace HelloWorld
+              {
+                [AliasedAttribute]
+                public class MyType {
+                    public int PublicPropertyInt { get; set; }
+                    public string PublicPropertyString { get; set; }
+                    private int PrivatePropertyInt { get; set; }
+                    private string PrivatePropertyString { get; set; }
 
-        private GeneratorDriver CreateDriver(Compilation compilation, params ISourceGenerator[] generators)
-            => new CSharpGeneratorDriver(
-                 new CSharpParseOptions(kind: SourceCodeKind.Regular, documentationMode: DocumentationMode.Parse),
-                ImmutableArray.Create(generators),
-                ImmutableArray<AdditionalText>.Empty);
+                    public double PublicDouble;
+                    public char PublicChar;
+                    private double PrivateDouble;
+                    private char PrivateChar;
 
-        private Compilation RunGenerators(Compilation compilation, out ImmutableArray<Diagnostic> diagnostics, params ISourceGenerator[] generators)
-        {
-            CreateDriver(compilation, generators).RunFullGeneration(compilation, out Compilation outCompilation, out diagnostics);
-            return outCompilation;
+                    public void MyMethod() { }
+                    public void MySecondMethod() { }
+                }
+
+                [AliasedAttribute(typeof(JsonConverterAttribute))]
+                public class NotMyType { }
+
+                [JsonSerializable(typeof(JsonConverterAttribute))]
+                public class ShouldNotFind { }
+
+              }";
+
+            Compilation compilation = CompilationHelper.CreateCompilation(source);
+
+            JsonSerializerSourceGenerator generator = new JsonSerializerSourceGenerator();
+
+            Compilation outCompilation = CompilationHelper.RunGenerators(compilation, out var generatorDiags, generator);
+
+            // Check base functionality of found types.
+            Assert.Equal(2, generator.foundTypes.Count);
+
+            // Check for MyType.
```

Comment:
Looks like there's common test logic (i.e verifying properties, fields, etc) between this and the above tests. Can we extract into a helper?